docs/source/datasources/glofas.rst

```diff
@@ -0,0 +1,315 @@
+
+Mac and Linux: Create the file `~/.cdsapirc`.
+
+Then, for all operating systems, add the following to the created file:
+
+.. code-block:: shell
+
+    url: https://cds.climate.copernicus.eu/api/v2
+    key: $UID:$API-key
+
+where `$UID` is your UID and `$API-key` is your API key.
+
+Next, if it :ref:`doesn't already exist<list of supported countries>`,
+you need to create a country configuration
+for the country you would like to analyze.
+
+An example country config for Bangladesh is:
+
+.. code-block:: yaml
+
+    iso3: bgd
+    glofas:
+      reporting_points:
+      - name: Bahadurabad
+        lon: 89.65
+        lat: 25.15
+      - name: Hardinge Bridge
+        lon: 89.05
+        lat: 24.05
+
+The reporting points indicate the raster file coordinates used
+to extract the river discharge for a particular location.
+Those in the above example have been taken from the
+`GloFAS map viewer
+<https://www.globalfloods.eu/glofas-forecasting/>`_.
+If you select "Reporting Points" from the "Hydrological"
+menu at the top, they will appear as dots on the map. If you then
+click on one of the points, you are able to see
+information such as the station name, and LISFLOOD X and Y, which are
+the respective longitude and latitude used in the configuration file.
+
+Reporting point coordinates are manually selected by the GloFAS team to
+be representative of physical gauge locations, and to be located on a river
+in the model raster file. In principle, one could
+specify any set of coordinates that exists on the raster, but caution is advised
+when doing so.
+
+You can initialize a built-in country config as follows:
+
+.. code-block:: python
+
+    from aatoolbox import create_country_config
+
+    country_config = create_country_config(iso3="bgd")
+
+Another required input is the geographic area of interest, which will
+define the bounds of raw raster data to be downloaded. A simple
+way to identify the area around the chosen country is to use the COD
+administrative boundaries. You will need to download the data,
+and extract a geo bounding box:
+
+.. code-block:: python
+
+    from aatoolbox import CodAB, GeoBoundingBox
+
+    codab = CodAB(country_config=country_config)
+    codab.download()
+    admin0 = codab.load()
+    geo_bounding_box = GeoBoundingBox.from_shape(admin0)
+
+Note that the reporting points in the configuration file need to lie within
+the geographic area of interest.
+
+Next you need to instantiate the GloFAS class with the country config. For this
+example, we will use the GloFAS forecast, however the steps are similar for the
+the reanalysis and reforecast.
+
+In this case, we would like the data for the past month. In general, we suggest
+specifying dates explicitly, as using e.g. `datetime.today()` will not produce
+the same results when run on a different day.
+
+.. code-block:: python
+
+    from datetime import date
+
+    from aatoolbox import GlofasForecast
+
+    glofas_forecast = GlofasForecast(
```

Comment:
It would be good to have some sample code also for the other two classes, or at least mention what changes when using them. For instance, the class `GlofasReanalysis` does not accept `leadtime_max` as an argument.

Reply:
Yes was also being lazy here, will add the other examples